warehouse/internal/repo/upload.go

```diff
@@ -341,6 +383,18 @@ func (uploads *Uploads) UploadJobsStats(ctx context.Context, destType string, op
 	return stats, nil
 }
 
+func (uploads *Uploads) DeleteWaiting(ctx context.Context, uploadID int64) error {
+	_, err := uploads.db.ExecContext(ctx,
```

Comment:
Initially, I thought we might need to reset the `upload_id` as well from the `staging files`. But since we have `ON DELETE` clause.

Reply:
I should have mentioned, I have updated the PR description.

We would also need an index on upload_id to ensure the deletion won't be costly.